hip/CMakeLists.txt

```diff
@@ -130,34 +127,11 @@ if (GINKGO_HAVE_ROCTX)
     target_link_libraries(ginkgo_hip PRIVATE roc::roctx)
 endif()
 
-if(GINKGO_HIP_PLATFORM MATCHES "${HIP_PLATFORM_AMD_REGEX}")
-    find_package(hip REQUIRED)
-    # To save a bit of pain, we directly link against the `library` instead of
-    # linking against the target.
-    if (CMAKE_BUILD_TYPE)
-        # Check if our configuration is available first
-        string(TOUPPER "${CMAKE_BUILD_TYPE}" UPPER_BUILD_TYPE)
-        get_target_property(HIP_LIBAMDHIP64_LIBRARIES hip::amdhip64 IMPORTED_LOCATION_${UPPER_BUILD_TYPE})
-    endif()
-    if (NOT HIP_LIBAMDHIP64_LIBRARIES)
-        # Fall back to anything
-        get_target_property(HIP_LIBAMDHIP64_LIBRARIES hip::amdhip64 LOCATION)
-    endif()
-    target_link_libraries(ginkgo_hip PUBLIC ${HIP_LIBAMDHIP64_LIBRARIES})
-elseif(GINKGO_HIP_PLATFORM MATCHES "${HIP_PLATFORM_NVIDIA_REGEX}")
-    find_package(CUDA 10.1 REQUIRED)
-    target_link_libraries(ginkgo_hip PUBLIC ${CUDA_LIBRARIES})
-endif()
+target_compile_options(ginkgo_hip PRIVATE $<$<COMPILE_LANGUAGE:CXX>:${GINKGO_COMPILER_FLAGS}>)
 
 # Try to find everything in /opt/rocm/lib first.
 set(GKO_HIP_RPATH "${ROCM_PATH}/lib" )
```

Comment:
are these still necessary?

Reply:
I wouldn't think so, I'll try to remove them